### ISSUE TITLE
Cannot delete message from IMPAP account.

### ISSUE THREAD

**Issue by louisoft88** [jstedfast/MailKit#1738]
I use .Net 4.8 and MimeKit 4.4.0.0.

The following code does not delete any email from my IMAP account.
```c#
private async Task<int> DeleteSelectEmailsFromAccount(Account account)
{
...
...
   using (var client = new ImapClient())
  {
      client.Connect(account.HostName, account.Port, account.UseSsl);
      client.Authenticate(account.Email, account.Password);

      var inbox = client.Inbox;
      await inbox.OpenAsync(FolderAccess.ReadWrite);

      inbox.AddFlags(emailListItem.EmailMessage.Uid, MessageFlags.Deleted, true);
      inbox.Expunge();
  }

  client.Disconnect(true);
}
```

How can I delete an email from an IMAP account's inbox?
Thanks,
Louis

**Comment by jstedfast** [comment]
Why is client.Disconnect(true) outside of the using block?

Why are you using inbox.OpenAsync() but all other methods you call the sync API?

The AddFlags() and Expunge() should delete the message. If they don't, I'll need a protocol log to diagnose the issue.

**Comment by louisoft88** [comment]
Thanks for the very quick reply. 

(Congratulations! MailKit is an excellent tool!!!)

Async: I could not delete the message so I tried everything. THanks for notify me.

Explunge(): Typo, sorry I just wanted to make the code more readable.

I attached a log file, when I wanted to delete a message. 

When I delete a message then it disappears from my email list on the screen (I remove it) but getting messages next time shows the "deleted" message again. I checked it with my webmail app. The "deleted" 
[ImapLog.txt](https://github.com/jstedfast/MailKit/files/14765243/ImapLog.txt)
file was not deleted.

Many thanks,
Louis
